src/realm/views.hpp

```diff
@@ -56,20 +56,53 @@ class SortDescriptor {
     {
         return !m_columns.empty();
     }
-
-    // handover support
-    using HandoverPatch = std::unique_ptr<SortDescriptorHandoverPatch>;
-    static void generate_patch(SortDescriptor const&, HandoverPatch&);
-    static SortDescriptor create_from_and_consume_patch(HandoverPatch&, Table const&);
-
+    // returns whether this descriptor has any custom ascending/descending order
+    bool has_custom_order() const noexcept
+    {
+        return std::any_of(m_ascending.begin(), m_ascending.end(), [](bool b) { return !b; });
+    }
+    void merge_with(SortDescriptor&& other);
     class Sorter;
     Sorter sorter(IntegerColumn const& row_indexes) const;
 
+    // handover support
+    std::vector<std::vector<size_t>> export_column_indices() const;
+    std::vector<bool> export_order() const;
 private:
     std::vector<std::vector<const ColumnBase*>> m_columns;
     std::vector<bool> m_ascending;
 };
 
+// Distinct uses the same syntax as sort except that the order is meaningless.
```

Comment:
Yes, good observation. I also considered this design, but for simplicity I decided not to go that route. It would have increased the complexity because we would be dealing with allocations on the heap and manual memory management can open up more possibilities for mistakes, plus dealing with handover would be more tedious.